### PR TITLE
feat(source/cloud-storage): add bucket management tools

### PR DESCRIPTION
**Pull request by huangjiahua**
## 1. Description

Adds the final Cloud Storage bucket administration tools:

- `cloud-storage-create-bucket`
- `cloud-storage-get-bucket-metadata`
- `cloud-storage-get-bucket-iam-policy`
- `cloud-storage-delete-bucket`

The implementation adds source-level bucket operations, tool registration, unit tests, integration coverage, and documentation for the new tools.

Also wires up a `cloud-storage` prebuilt config (`--prebuilt cloud-storage`) that exposes all 14 Cloud Storage tools through two toolsets — `cloud-storage-buckets` (bucket admin) and `cloud-storage-objects` (object management) — driven by a `CLOUD_STORAGE_PROJECT` env var, with matching prebuilt-configs docs.

## 2. PR Checklist

- [x] Make sure to open an issue as a bug/issue before writing your code!
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)
- [x] Make sure to add `!` if this involves a breaking change

## 3. Issue Reference

Draft PR; issue reference TBD. 🦕

### CODE REVIEW

File: docs/en/integrations/cloud-storage/tools/cloud-storage-create-bucket.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+title: "cloud-storage-create-bucket"
```
[the remark this answers is not in the document]
Comment (huangjiahua) [reply docs/en/integrations/cloud-storage/tools/cloud-storage-create-bucket.md:2]: No, I didn't add this to other tools.

File: docs/en/integrations/cloud-storage/tools/cloud-storage-delete-bucket.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+---
+title: "cloud-storage-delete-bucket"
```
[the remark this answers is not in the document]
Comment (huangjiahua) [reply docs/en/integrations/cloud-storage/tools/cloud-storage-delete-bucket.md:2]: No, I didn't add this to other tools.

File: docs/en/integrations/cloud-storage/tools/cloud-storage-get-bucket-iam-policy.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+---
+title: "cloud-storage-get-bucket-iam-policy"
```
[the remark this answers is not in the document]
Comment (huangjiahua) [reply docs/en/integrations/cloud-storage/tools/cloud-storage-get-bucket-iam-policy.md:2]: No, I didn't add this to other tools.

File: docs/en/integrations/cloud-storage/tools/cloud-storage-get-bucket-metadata.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+---
+title: "cloud-storage-get-bucket-metadata"
```
[the remark this answers is not in the document]
Comment (huangjiahua) [reply docs/en/integrations/cloud-storage/tools/cloud-storage-get-bucket-metadata.md:2]: No, I didn't add this to other tools.

File: internal/prebuiltconfigs/tools/cloud-storage.yaml
Comment (Yuan325) [inline internal/prebuiltconfigs/tools/cloud-storage.yaml:1]: Please add the unit test to `TestPrebuiltTools()` in `cmd/internal/config_test.go`

https://github.com/googleapis/mcp-toolbox/blob/9910ce574f6b1e438bd8bb97a1c2dfdf50d19b02/cmd/internal/config_test.go#L1424

File: internal/sources/cloudstorage/cloudstorage.go [hunk 5]
```diff
@@ -219,6 +220,81 @@ func (s *Source) ListBuckets(ctx context.Context, project, prefix string, maxRes
 	}, nil
 }
 
+// CreateBucket creates a Cloud Storage bucket in the source project and returns
+// its freshly-read metadata. When location is empty, Cloud Storage defaults to
+// "US"; the tool layer also supplies that default for clearer manifests.
+func (s *Source) CreateBucket(ctx context.Context, bucket, location string, uniformBucketLevelAccess bool) (map[string]any, error) {
+	if location == "" {
+		location = "US"
+	}
```
Comment (Yuan325) [inline internal/sources/cloudstorage/cloudstorage.go]: Will not need this since we already set the default. It will be resolved during`ParseParam()` before the `Invoke()` function is called.
```suggestion
```

File: internal/tools/cloudstorage/cloudstoragecreatebucket/cloudstoragecreatebucket.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+	if err != nil {
+		return nil, util.NewClientServerError("source used is not compatible with the tool", http.StatusInternalServerError, err)
+	}
+
+	mapParams := params.AsMap()
+	bucket, ok := mapParams[bucketKey].(string)
+	if !ok || bucket == "" {
+		return nil, util.NewAgentError(fmt.Sprintf("invalid or missing '%s' parameter; expected a non-empty string", bucketKey), nil)
+	}
+	location, _ := mapParams[locationKey].(string)
```
Comment (Yuan325) [inline internal/tools/cloudstorage/cloudstoragecreatebucket/cloudstoragecreatebucket.go:114]: If you want to check for the location to make sure it's not "", you can check it here. The default should be in effect at this stage.

File: tests/cloudstorage/cloud_storage_integration_test.go [hunk 7]
```diff
@@ -462,6 +546,12 @@ func TestCloudStorageToolEndpoints(t *testing.T) {
 	runCloudStorageReadObjectTest(t, bucketName)
 	runCloudStorageListBucketsTest(t, bucketName)
 	runCloudStorageGetObjectMetadataTest(t, bucketName)
+	bucketToolName := "toolbox-it-bucket-" + strings.ReplaceAll(uuid.New().String(), "-", "")[:17]
+	defer cleanupGCSBucket(ctx, t, client, bucketToolName)
```
Comment (Yuan325) [inline tests/cloudstorage/cloud_storage_integration_test.go:549]: is this a failsafe incase the test halt before `runCloudStorageDeleteBucketTest()`?

**Review by Yuan325 — CHANGES_REQUESTED** [review]
Thank you! Left some feedbacks~

File: tests/cloudstorage/cloud_storage_integration_test.go (hunk 7 above)
Comment (huangjiahua) [reply tests/cloudstorage/cloud_storage_integration_test.go:549]: Yes, it is.

File: internal/tools/cloudstorage/cloudstoragecreatebucket/cloudstoragecreatebucket.go (hunk 6 above)
Comment (huangjiahua) [reply internal/tools/cloudstorage/cloudstoragecreatebucket/cloudstoragecreatebucket.go:114]: I made the default location "" and removed all location checking. GCS server will have a default. Also added integration test cases with empty location param.

File: internal/sources/cloudstorage/cloudstorage.go (hunk 5 above)
Comment (huangjiahua) [reply internal/sources/cloudstorage/cloudstorage.go]: Removed.

File: internal/prebuiltconfigs/tools/cloud-storage.yaml
Comment (huangjiahua) [reply internal/prebuiltconfigs/tools/cloud-storage.yaml:1]: Added.

**Review by Yuan325 — APPROVED** [review]
